static/js/ajax-all-builds.js

```diff
@@ -0,0 +1,62 @@
+$(function () {
+    $('.filter').change(function () {
+        $.ajax({
+            type: 'POST',
+            url: '/all_builds/filter/',
+            data: {
+                'builder': $('#builder-filter').val(),
+                'status': $('#status-filter').val(),
+                'page': 1,
+                'csrfmiddlewaretoken': $("input[name=csrfmiddlewaretoken]").val()
+            },
+            success: filterSuccess,
+            beforeSend: function () {
+                $('#all_builds_table').html("Applying filter...")
+            },
+            dataType: 'html'
+        });
+    });
+});
+
+$(function () {
+    $('.filter').ready(function () {
+        $.ajax({
+            type: 'POST',
+            url: '/all_builds/filter/',
+            data: {
+                'builder': $('#builder-filter').val(),
+                'status': $('#status-filter').val(),
+                'page': 1,
+                'csrfmiddlewaretoken': $("input[name=csrfmiddlewaretoken]").val()
+            },
+            success: filterSuccess,
+            beforeSend: function () {
+                $('#all_builds_table').html("Loading Builds...")
+            },
+            dataType: 'html'
+        });
+    });
+});
+
+function changePage(page) {
+    console.log("Got you");
+    $.ajax({
+        type: 'POST',
+        url: '/all_builds/filter/',
+        data: {
+            'builder': $('#builder-filter').val(),
+            'status': $('#status-filter').val(),
+            'page': page,
+            'csrfmiddlewaretoken': $("input[name=csrfmiddlewaretoken]").val()
```

Comment:
What does this do?

Reply:
This needs to be passed with all POST requests- built in support for additional security.